queries/dune_v2/period_slippage.sql

```diff
@@ -183,185 +183,51 @@ incoming_and_outgoing as (
-                            )
-                          else
-                              case
-                                  when a.usd_value is not null and b.usd_value is not null
-                                      -- If prices from the prices.usd dune table are used, the prices can also be off from time to time.
-                                      -- On the one hand side, we don't wanna allow to high deviations. E.g. for
-                                      -- 0x9a318d1abd997bcf8afed55b2946a7b1bd919d227f094cdcc99d8d6155808d7c a matchability of 0.26 is calculated
-                                      -- for a slippage of WETH and the LDO deficit. (In this example the internal trade is only STRONG -> LDO)
-                                      -- On the other hand, real internal trades like the CRV to USDT internal trade of 0xc15dda7c10eb317c0ad177316020ec4baa13babb0713b73480feef14045603f4
-                                      -- also score a matchablilty of 0.027
-                                      -- As a compromise 0.025 was chosen
-                                      then (abs((a.usd_value + b.usd_value) /
-                                                (abs(a.usd_value) + abs(b.usd_value))) <
-                                            0.025
-                                      and a.token != b.token
-                                      and abs(a.usd_value) > 10 -- we don't want small slippage values to be recognized as internal swaps
-                                      )
-                                  else
-                                      false
-                                  end
-                          end
-                  else
-                      false
-                  end
-              )
+,block_range as (
+  select min(number) as start_block,
+       max(number) as end_block
+  from ethereum.blocks
+  where time between cast('{{StartTime}}' as timestamp) and cast('{{EndTime}}' as timestamp)
 )
-,incoming_and_outgoing_with_buffer_trades as (
-select * from (
-    select block_time,
-          tx_hash,
-          solver_address,
-          symbol,
-          token,
-          amount,
-          transfer_type
-    from incoming_and_outgoing
-    union all
-    select block_time,
-          tx_hash,
-          solver_address,
-          symbol,
-          token_from as token,
-          amount_from as amount,
-          transfer_type
-    from buffer_trades
-) as _
-order by block_time
+,internalized_imbalances as (
+  select  b.block_time,
+          b.tx_hash,
+          b.solver_address,
+          t.symbol,
+          from_hex(i.token) as token,
+          cast(cast(i.amount as varchar) as int256) as amount,
+          'PHANTOM_TRANSFER' as transfer_type
+    from cowswap.raw_internal_imbalance i
+    inner join cow_protocol_ethereum.batches b
+        on i.block_number = b.block_number
+        and from_hex(i.tx_hash) = b.tx_hash
+    join tokens.erc20 t
```

Comment:
Thanks for pointing this out. I believe we need to skip that token because we wouldn't have its decimals. However, its probably best not to skip it too early so we can actually identify when a calculation is off due to missing tokens. That is, we hold on the the WEI amounts as long as possible. Eventually we can move this calculation into our own database and fail when we encounter missing tokens.

I hope this clears up your question.

Reply:
Good point. I think it would be nice to have a collection of these missed tokens in some CTE so we can keep track of them for each solver payout. What do you think about that?

Is not urgent, but let's move the potential discussion to a separate issue if we decide to move forward with this.